### ISSUE TITLE
Duplicate UUIDs - V5 inconsistencies when using UTF8/Unicode characters

### ISSUE THREAD

**Issue by vjamrich** [daegalus/dart-uuid#111]
Two different input names can result in the same v5 UUID - e.g. `人` & `äºº` will both be encoded to `f7cd5579-8db4-5043-bd83-98565a62a4f4`
This is caused by the v5 not working properly with UTF8 characters.

### Dart
```
import "package:uuid/uuid.dart";
final String key = const Uuid().v5(Uuid.NAMESPACE_X500, "人");
print(key);
```
Output: `f7cd5579-8db4-5043-bd83-98565a62a4f4`

### Python
```
import uuid
key = uuid.uuid5(
    namespace=uuid.NAMESPACE_X500,
    name="人",
)
print(key)
```
Output: `3af28677-8030-59ef-81a6-051ddf226fad`

### Expected Output
The output String should be consistent.

**Comment by vjamrich** [comment]
@daegalus created the following pull request: #112

**Comment by vjamrich** [comment]
Additionally, this means two different input names can result in the same UUID - e.g. "人" & "äºº" will both

**Comment by daegalus** [comment]
I merged the PR. good catch. Did not conisder UTF8. I will add some tests to cover regressions.

**Comment by daegalus** [comment]
The fix is now part of the v4.4.0 release, sorry for the delay, there were other problems I had to fix due other problems fixed as part of the release.
